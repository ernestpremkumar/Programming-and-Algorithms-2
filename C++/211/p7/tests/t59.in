breadth
print
breadth
echo 7
find 7
breadth
find 2
size
breadth
breadth
balanced
size
insert 6
insert 7
insert 6
echo 5
print
distance
echo 0
size
balanced
print
insert 5
insert 1
print
print
size
breadth
insert 3
size
find 3
size
find 0
echo 0
print
balanced
distance
echo 3
print
breadth
insert 1
echo 1
print
balanced
find 5
echo 1
print
distance
find 7
size
echo 7
find 2
breadth
breadth
breadth
find 3
print
size
echo 0
distance
find 3
insert 6
insert 7
distance
distance
find 1
find 7
breadth
size
size
distance
breadth
echo 2
find 0
size
find 1
balanced
breadth
breadth
print
print
size
distance
echo 0
find 0
balanced
echo 1
find 5
breadth
insert 6
balanced
print
insert 7
balanced
breadth
size
print
insert 4
print
find 1
find 7
echo 2
balanced
print
size
find 5
insert 2
insert 0
insert 3
insert 0
print
echo 3
echo 0
find 1
size
print
echo 5
insert 4
echo 0
print
size
breadth
echo 3
distance
distance
echo 0
find 3
find 0
insert 3
echo 6
print
echo 4
echo 5
breadth
distance
print
insert 0
print
balanced
find 2
size
echo 3
find 6
insert 4
size
balanced
breadth
print
insert 6
balanced
size
echo 3
print
breadth
distance
echo 0
find 1
balanced
breadth
size
size
distance
insert 7
breadth
find 2
insert 4
balanced
size
breadth
distance
echo 4
breadth
echo 3
size
print
print
distance
distance
breadth
insert 3
echo 1
balanced
print
find 1
echo 1
breadth
echo 0
print
distance
breadth
size
insert 2
breadth
balanced
find 2
find 0
find 3
size
find 7
echo 5
balanced
print
echo 7
echo 0
distance
size
balanced
insert 3
find 4
balanced
insert 6
insert 2
breadth
find 3
insert 0
print
size
balanced
insert 2
insert 7
breadth
insert 3
balanced
insert 2
find 2
echo 0
size
insert 1
print
echo 5
insert 5
breadth
breadth
breadth
print
balanced
size
distance
size
find 7
balanced
find 3
print
insert 4
insert 1
distance
breadth
breadth
insert 3
size
size
balanced
insert 4
distance
print
insert 1
breadth
print
insert 1
size
distance
breadth
breadth
balanced
breadth
find 1
size
print
balanced
distance
find 3
distance
print
size
echo 5
echo 4
breadth
find 5
balanced
echo 2
find 5
print
distance
echo 0
size
insert 6
distance
distance
find 3
echo 0
breadth
size
breadth
breadth
balanced
size
insert 2
balanced
balanced
distance
distance
balanced
balanced
balanced
distance
distance
echo 3
distance
distance
find 3
echo 5
echo 5
find 2
size
balanced
print
balanced
insert 3
insert 7
insert 1
insert 0
find 7
echo 1
balanced
print
find 2
balanced
balanced
find 3
breadth
insert 2
balanced
breadth
balanced
insert 6
distance
echo 2
print
find 0
distance
distance
breadth
breadth
echo 4
breadth
size
balanced
distance
size
find 6
balanced
distance
find 7
echo 2
breadth
balanced
echo 3
insert 4
echo 6
distance
balanced
find 1
size
find 1
balanced
echo 6
echo 6
distance
echo 3
size
print
insert 1
breadth
find 2
find 5
size
size
size
balanced
insert 5
print
breadth
balanced
balanced
insert 7
breadth
find 5
size
echo 1
distance
balanced
distance
distance
distance
distance
print
insert 2
distance
print
echo 5
balanced
size
find 3
print
size
balanced
find 1
find 5
distance
balanced
insert 4
distance
print
print
insert 0
breadth
size
print
breadth
insert 0
size
insert 0
breadth
echo 4
breadth
balanced
find 7
size
insert 7
echo 6
size
print
breadth
size
echo 2
size
find 2
echo 3
balanced
print
echo 4
find 3
insert 3
size
find 2
breadth
echo 6
size
insert 5
echo 4
balanced
print
breadth
find 4
breadth
echo 7
size
echo 4
insert 3
distance
breadth
find 3
breadth
size
size
echo 2
print
print
echo 1
breadth
distance
size
size
print
breadth
echo 2
breadth
balanced
insert 2
insert 1
print
print
breadth
size
balanced
find 7
breadth
distance
echo 7
find 2
print
balanced
balanced
print
balanced
find 5
distance
print
echo 3
echo 6
breadth
breadth
insert 3
size
breadth
distance
insert 1
size
find 0
echo 6
find 2
distance
size
print
echo 2
size
echo 5
echo 7
size
size
echo 7
echo 0
breadth
insert 2
insert 0
find 1
print
distance
find 7
insert 2
balanced
echo 3
size
insert 1
print
find 4
print
distance
distance
size
insert 7
find 5
distance
print
print
print
insert 3
find 5
distance
echo 5
insert 2
distance
balanced
balanced
print
print
balanced
size
insert 6
balanced
breadth
insert 4
insert 6
size
insert 7
insert 3
balanced
echo 7
print
print
insert 0
breadth
echo 1
balanced
distance
balanced
size
size
breadth
size
balanced
insert 3
find 6
breadth
echo 7
balanced
insert 2
find 4
find 2
print
echo 2
breadth
print
distance
print
balanced
print
size
insert 7
distance
echo 2
print
insert 1
find 7
insert 5
balanced
find 1
print
print
echo 3
print
find 7
echo 0
echo 5
print
find 6
print
distance
distance
size
insert 7
insert 1
distance
find 5
echo 2
size
size
breadth
find 2
print
find 7
print
balanced
breadth
find 5
breadth
insert 4
echo 6
find 7
insert 3
distance
echo 3
echo 1
find 5
size
find 7
balanced
breadth
insert 6
find 2
size
insert 0
distance
size
breadth
distance
distance
print
print
size
balanced
print
print
print
print
insert 0
echo 6
insert 2
print
insert 3
distance
print
breadth
find 1
insert 1
echo 5
size
balanced
distance
find 3
find 3
insert 4
balanced
find 3
distance
print
find 5
insert 0
insert 0
size
insert 2
find 2
distance
insert 2
breadth
echo 4
insert 1
breadth
print
find 0
print
echo 7
balanced
balanced
distance
find 1
find 7
find 6
echo 2
balanced
insert 6
balanced
echo 2
size
print
echo 5
print
balanced
insert 3
print
echo 7
balanced
distance
breadth
distance
breadth
find 7
insert 4
echo 4
insert 3
echo 4
print
insert 4
echo 3
insert 2
size
print
breadth
find 3
size
balanced
insert 7
find 2
distance
distance
balanced
echo 6
find 3
distance
print
balanced
insert 6
balanced
balanced
echo 2
find 1
echo 7
size
balanced
distance
echo 5
find 5
echo 0
size
insert 6
breadth
print
breadth
print
breadth
breadth
print
breadth
echo 6
balanced
insert 6
find 7
balanced
find 3
insert 1
balanced
breadth
breadth
distance
distance
distance
print
echo 3
insert 1
breadth
distance
balanced
size
print
echo 5
size
print
find 7
size
size
breadth
print
print
size
breadth
distance
breadth
breadth
print
breadth
balanced
breadth
balanced
print
size
insert 6
balanced
insert 5
size
insert 4
find 7
size
print
insert 7
insert 1
distance
distance
find 3
find 5
distance
breadth
distance
breadth
echo 7
find 7
distance
echo 7
echo 0
echo 5
distance
balanced
insert 4
balanced
size
find 2
echo 0
echo 0
breadth
distance
print
balanced
echo 2
size
breadth
insert 2
find 3
balanced
distance
echo 6
echo 1
insert 3
print
balanced
distance
balanced
balanced
print
balanced
echo 7
insert 4
balanced
size
insert 6
print
size
breadth
breadth
breadth
find 5
size
distance
insert 1
insert 2
insert 0
print
size
echo 3
insert 5
distance
find 5
size
find 1
size
print
print
echo 7
echo 2
print
size
size
insert 6
size
size
size
insert 0
echo 6
balanced
size
print
balanced
distance
size
echo 6
echo 6
print
insert 2
balanced
distance
distance
insert 1
balanced
size
breadth
balanced
print
insert 6
breadth
echo 3
insert 6
echo 0
find 6
find 1
print
print
find 3
size
balanced
echo 5
size
breadth
distance
echo 6
insert 3
insert 0
insert 2
echo 2
find 4
distance
distance
find 2
echo 1
find 5
size
find 3
size
size
distance
print
find 7
insert 5
balanced
echo 2
distance
find 6
insert 4
breadth
find 1
echo 0
find 6
size
find 5
distance
balanced
echo 2
balanced
distance
find 7
size
breadth
breadth
find 6
balanced
size
print
size
find 7
print
distance
find 3
distance
balanced
insert 4
find 3
print
echo 0
distance
print
size
