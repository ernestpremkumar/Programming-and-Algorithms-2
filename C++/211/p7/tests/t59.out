{}
{}
{}
7
<7> is not in tree.
{}
<2> is not in tree.
0
{}
{}
Tree is balanced.
0
5
{6, 7}
Average distance of nodes to root = 0.5
0
2
Tree is balanced.
{6, 7}
{1, 5, 6, 7}
{1, 5, 6, 7}
4
{6, 5, 7, 1}
5
<3> is in tree.
5
<0> is not in tree.
0
{1, 3, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 1.4
3
{1, 3, 5, 6, 7}
{6, 5, 7, 1, 3}
1
{1, 3, 5, 6, 7}
Tree is not balanced.
<5> is in tree.
1
{1, 3, 5, 6, 7}
Average distance of nodes to root = 1.4
<7> is in tree.
5
7
<2> is not in tree.
{6, 5, 7, 1, 3}
{6, 5, 7, 1, 3}
{6, 5, 7, 1, 3}
<3> is in tree.
{1, 3, 5, 6, 7}
5
0
Average distance of nodes to root = 1.4
<3> is in tree.
Average distance of nodes to root = 1.4
Average distance of nodes to root = 1.4
<1> is in tree.
<7> is in tree.
{6, 5, 7, 1, 3}
5
5
Average distance of nodes to root = 1.4
{6, 5, 7, 1, 3}
2
<0> is not in tree.
5
<1> is in tree.
Tree is not balanced.
{6, 5, 7, 1, 3}
{6, 5, 7, 1, 3}
{1, 3, 5, 6, 7}
{1, 3, 5, 6, 7}
5
Average distance of nodes to root = 1.4
0
<0> is not in tree.
Tree is not balanced.
1
<5> is in tree.
{6, 5, 7, 1, 3}
Tree is not balanced.
{1, 3, 5, 6, 7}
Tree is not balanced.
{6, 5, 7, 1, 3}
5
{1, 3, 5, 6, 7}
{1, 3, 4, 5, 6, 7}
<1> is in tree.
<7> is in tree.
2
Tree is not balanced.
{1, 3, 4, 5, 6, 7}
6
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
3
0
<1> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
5
0
{0, 1, 2, 3, 4, 5, 6, 7}
8
{6, 5, 7, 1, 0, 3, 2, 4}
3
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
0
<3> is in tree.
<0> is in tree.
6
{0, 1, 2, 3, 4, 5, 6, 7}
4
5
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<2> is in tree.
8
3
<6> is in tree.
8
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
3
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
0
<1> is in tree.
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
8
8
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
<2> is in tree.
Tree is not balanced.
8
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
4
{6, 5, 7, 1, 0, 3, 2, 4}
3
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
1
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
1
{6, 5, 7, 1, 0, 3, 2, 4}
0
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
8
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
<2> is in tree.
<0> is in tree.
<3> is in tree.
8
<7> is in tree.
5
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
7
0
Average distance of nodes to root = 2.25
8
Tree is not balanced.
<4> is in tree.
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
<2> is in tree.
0
8
{0, 1, 2, 3, 4, 5, 6, 7}
5
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
Average distance of nodes to root = 2.25
8
<7> is in tree.
Tree is not balanced.
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
8
8
Tree is not balanced.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
<1> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.25
<3> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
8
5
4
{6, 5, 7, 1, 0, 3, 2, 4}
<5> is in tree.
Tree is not balanced.
2
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
0
8
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
<3> is in tree.
0
{6, 5, 7, 1, 0, 3, 2, 4}
8
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
8
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
3
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
<3> is in tree.
5
5
<2> is in tree.
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<7> is in tree.
1
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
Tree is not balanced.
Tree is not balanced.
<3> is in tree.
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
Average distance of nodes to root = 2.25
2
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
4
{6, 5, 7, 1, 0, 3, 2, 4}
8
Tree is not balanced.
Average distance of nodes to root = 2.25
8
<6> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.25
<7> is in tree.
2
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
3
6
Average distance of nodes to root = 2.25
Tree is not balanced.
<1> is in tree.
8
<1> is in tree.
Tree is not balanced.
6
6
Average distance of nodes to root = 2.25
3
8
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
<2> is in tree.
<5> is in tree.
8
8
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
<5> is in tree.
8
1
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
5
Tree is not balanced.
8
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
<1> is in tree.
<5> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
8
{6, 5, 7, 1, 0, 3, 2, 4}
4
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
<7> is in tree.
8
6
8
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
8
2
8
<2> is in tree.
3
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
4
<3> is in tree.
8
<2> is in tree.
{6, 5, 7, 1, 0, 3, 2, 4}
6
8
4
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
<4> is in tree.
{6, 5, 7, 1, 0, 3, 2, 4}
7
8
4
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
<3> is in tree.
{6, 5, 7, 1, 0, 3, 2, 4}
8
8
2
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
1
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
2
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
8
Tree is not balanced.
<7> is in tree.
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
7
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<5> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
3
6
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
8
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
8
<0> is in tree.
6
<2> is in tree.
Average distance of nodes to root = 2.25
8
{0, 1, 2, 3, 4, 5, 6, 7}
2
8
5
7
8
8
7
0
{6, 5, 7, 1, 0, 3, 2, 4}
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<7> is in tree.
Tree is not balanced.
3
8
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
8
<5> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
Average distance of nodes to root = 2.25
5
Average distance of nodes to root = 2.25
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
8
Tree is not balanced.
7
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
1
Tree is not balanced.
Average distance of nodes to root = 2.25
Tree is not balanced.
8
8
{6, 5, 7, 1, 0, 3, 2, 4}
8
Tree is not balanced.
<6> is in tree.
{6, 5, 7, 1, 0, 3, 2, 4}
7
Tree is not balanced.
<4> is in tree.
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
2
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 2.25
2
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
Tree is not balanced.
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
3
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
0
5
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
8
Average distance of nodes to root = 2.25
<5> is in tree.
2
8
8
{6, 5, 7, 1, 0, 3, 2, 4}
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
<5> is in tree.
{6, 5, 7, 1, 0, 3, 2, 4}
6
<7> is in tree.
Average distance of nodes to root = 2.25
3
1
<5> is in tree.
8
<7> is in tree.
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
<2> is in tree.
8
Average distance of nodes to root = 2.25
8
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
6
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
<1> is in tree.
5
8
Tree is not balanced.
Average distance of nodes to root = 2.25
<3> is in tree.
<3> is in tree.
Tree is not balanced.
<3> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
8
<2> is in tree.
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
4
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
7
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.25
<1> is in tree.
<7> is in tree.
<6> is in tree.
2
Tree is not balanced.
Tree is not balanced.
2
8
{0, 1, 2, 3, 4, 5, 6, 7}
5
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
7
Tree is not balanced.
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
<7> is in tree.
4
4
{0, 1, 2, 3, 4, 5, 6, 7}
3
8
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
<3> is in tree.
8
Tree is not balanced.
<2> is in tree.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Tree is not balanced.
6
<3> is in tree.
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
2
<1> is in tree.
7
8
Tree is not balanced.
Average distance of nodes to root = 2.25
5
<5> is in tree.
0
8
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
6
Tree is not balanced.
<7> is in tree.
Tree is not balanced.
<3> is in tree.
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
3
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
5
8
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
8
8
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
8
<7> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
<3> is in tree.
<5> is in tree.
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
{6, 5, 7, 1, 0, 3, 2, 4}
7
<7> is in tree.
Average distance of nodes to root = 2.25
7
0
5
Average distance of nodes to root = 2.25
Tree is not balanced.
Tree is not balanced.
8
<2> is in tree.
0
0
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
2
8
{6, 5, 7, 1, 0, 3, 2, 4}
<3> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.25
6
1
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.25
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
7
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
<5> is in tree.
8
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
8
3
Average distance of nodes to root = 2.25
<5> is in tree.
8
<1> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
7
2
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
8
8
8
6
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.25
8
6
6
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
Tree is not balanced.
8
{6, 5, 7, 1, 0, 3, 2, 4}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 5, 7, 1, 0, 3, 2, 4}
3
0
<6> is in tree.
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<3> is in tree.
8
Tree is not balanced.
5
8
{6, 5, 7, 1, 0, 3, 2, 4}
Average distance of nodes to root = 2.25
6
2
<4> is in tree.
Average distance of nodes to root = 2.25
Average distance of nodes to root = 2.25
<2> is in tree.
1
<5> is in tree.
8
<3> is in tree.
8
8
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
Tree is not balanced.
2
Average distance of nodes to root = 2.25
<6> is in tree.
{6, 5, 7, 1, 0, 3, 2, 4}
<1> is in tree.
0
<6> is in tree.
8
<5> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
2
Tree is not balanced.
Average distance of nodes to root = 2.25
<7> is in tree.
8
{6, 5, 7, 1, 0, 3, 2, 4}
{6, 5, 7, 1, 0, 3, 2, 4}
<6> is in tree.
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.25
<3> is in tree.
Average distance of nodes to root = 2.25
Tree is not balanced.
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
0
Average distance of nodes to root = 2.25
{0, 1, 2, 3, 4, 5, 6, 7}
8
